apps/dashboard/app/models/project.rb

```diff
@@ -91,6 +98,22 @@ def templates
         Project.new(**opts)
       end
     end
+
+    private
+
+    def importable_directories
+      Configuration.shared_projects_root.flat_map do |root|
+        next unless File.readable?(root)
+
+        Dir.each_child(root).flat_map do |child|
```

Comment:
It's not clear to me off the top why you're using `flat_map` here. I think maybe you should write some tests for this to be sure it's what you'd expect.  At a glance, this seems like it would return an array of String and Project objects.

Reply:
`flat_map` converts all the nested arrays into a 1D array (flattened); this by its [definition](https://medium.com/@harishreddyvantikommu/flat-map-an-unpopular-built-in-ruby-method-65ae5f20cb2f) is not a map. 
Also, I have tested out the function and it returns an array of projects.

Alternative could be to create a `project = []` array and fill it, and return it.

Reply:
> flat_map converts all the nested arrays into a 1D array (flattened); this by its [definition](https://medium.com/@harishreddyvantikommu/flat-map-an-unpopular-built-in-ruby-method-65ae5f20cb2f) is not a map.
Also, I have tested out the function and it returns an array of projects.

> Alternative could be to create a project = [] array and fill it, and return it.

I see, I generally just use `map { }.flatten`. Though I think `map` in this context (ruby) is more a synonym for `transform` (it's in fact an alias for `transform`) rather than map as a key value data structure. I.e., it's a verb not a noun.

Reply:
Yes, that is correct.